--- origsrc/Audio-TagLib-1.43/Makefile.PL	2007-01-16 20:49:27.000000000 -0600
+++ src/Audio-TagLib-1.43/Makefile.PL	2007-03-25 02:43:45.703125000 -0500
@@ -35,8 +35,8 @@
     LD                => 'g++',
     LDDLFLAGS         => $^O eq 'darwin' ? "-dynamiclib $libs" : "-shared ". $libs,
     XSOPT             => '-C++ -hiertype',
-    LIBS              => $libs, 
-    DEFINE            => $^O eq 'freebsd' ? '-D_BSD_ICONV -DNO_DOXYGEN' : '-DNO_DOXYGEN', # e.g., '-DHAVE_SOMETHING'
+    LIBS              => $^O =~ m/cygwin|freebsd/ ? "$libs -liconv" : $libs,
+    DEFINE            => $^O =~ m/cygwin|freebsd/ ? '-D_BSD_ICONV -DNO_DOXYGEN' : '-DNO_DOXYGEN',
     INC               => $inc, # e.g., '-I. -I/usr/include/other'
 	# Un-comment this if you add C files to link with later:
     # OBJECT            => '$(O_FILES)', # link all the C files too
